frontend/src/app/app.component.ts

```diff
@@ -1,10 +1,48 @@
-import { Component } from '@angular/core';
+import { AfterViewInit, Component, DestroyRef, OnInit } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Router } from '@angular/router';
+import { AuthService } from '@services/auth.service';
+import { ToastService } from '@services/toast.service';
+import { take } from 'underscore';
 
 @Component({
   selector: 'lc-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
-export class AppComponent {
-  title = 'lettercraft';
+export class AppComponent implements OnInit, AfterViewInit {
+    title = 'lettercraft';
+
+    constructor(
+        private authService: AuthService,
+        private router: Router,
+        private toastService: ToastService,
+        private destroyRef: DestroyRef,
+    ) { }
+
+    ngOnInit(): void {
+        this.authService.logoutResult$.pipe(
```

Comment:
I generally try to avoid subscribing in a service, but I agree with you that the `AppComponent` is not an obvious place to look for this subscription.

Can I offer you `user-menu.component.ts` as a compromise? :P (It's also the place where `logout$.next()` is called.)